### ISSUE TITLE
Learning rate compared to adamw

### ISSUE THREAD

**Issue by davnords** [KellerJordan/Muon#49]
Hi! I see the example code provides 67x increase of learning rate for muon params vs. adamw params. Is this a reasonable heuristic for translating previously used adamw learning rate to now using muon for training?

Thank you,

/ David

**Comment by kiriri** [comment]
Funnily enough I was just looking into the exact same thing, and I found a learning rate like you would use with adamw to be much more effective than the 0.1 from the example. In my case 1e-3 gave me an accuracy of 0.84 after 1 epoch, while the suggested rate of 1e-1 gave me only 0.36 on the same model/dataset.

**Comment by davnords** [comment]
Thank you!

**Comment by parlance-zz** [comment]
> Funnily enough I was just looking into the exact same thing, and I found a learning rate like you would use with adamw to be much more effective than the 0.1 from the example. In my case 1e-3 gave me an accuracy of 0.84 after 1 epoch, while the suggested rate of 1e-1 gave me only 0.36 on the same model/dataset. Even 10 epochs later, the 0.1 lr only reached acc 0.52

May I ask what kind of model you were training, param count and batch size?

**Comment by parlance-zz** [comment]
Trying to get the perfect learning rate schedule is driving me crazy.

I train music diffusion models using a customized EDM2 model architecture (demos [here](https://www.g-diffuser.com/dualdiffusion/), source [here](https://github.com/parlance-zz/dualdiffusion) if you're interested).

The EDM2 paper was all about improving training dynamics. As such it comes with force-normalized weights (no need to worry about weight decay) initialized from unit gaussian (desirable properties for random matrix singular value distribution), no biases whatsoever in the network, operations preserve unit variance on expectation as much as possible, QKV are all normalized per head, etc. There is definitely some synergy with the Muon optimizer.

The base learning rate for my EDM2 models with AdamW is 1e-2, which is quite high for AdamW, especially for models  >= 500m params. Nevertheless it consistently produces good results with EDM2, even with large models and the schedule has been rigorously tested and tuned.

From what I've seen so far a good base rate for Muon is around 50x the AdamW rate, an astonishing 0.5 LR(!). Train loss is wiped out absurdly quickly, and I was able to train models to the same loss as AdamW @ 300k steps with Muon @ 12k steps, an astonishing 25x speedup.

The problem isn't stability, but rather that the learn rate for reducing the most loss per training step decays in a very strange way that I haven't been able to pin down. Polynomial decay is what I used with AdamW, decay exponent of 0.5 works well (as in the EDM2 paper). With Muon the optimal decay is much faster with my setup, but I haven't had any luck trying to tune a static schedule. I've tried all kinds of decay rates.

I feel like there must be a way to dynamically determine at least an approximate optimal LR with the statistics we have but I haven't been successful in finding it. If an optimal schedule could be automated it would be possible to achieve this 25x speedup without the pain of trial and error, reloading checkpoints etc.
